sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/ParquetCompressionCodecMapper.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.execution.datasources.parquet;
+
+import java.util.Locale;
+
+import org.apache.parquet.hadoop.metadata.CompressionCodecName;
+
+/**
+ * A mapper class easy to obtain compression codecs based on their names.
+ */
+public enum ParquetCompressionCodecMapper {
+  NONE(null),
+  UNCOMPRESSED(CompressionCodecName.UNCOMPRESSED),
```

Comment:
Why make our own enum if there is already an enum-like list of codecs in parquet?

Reply:
One reason is Spark add the compression codecs `none`, another is out-of-date. Before https://github.com/apache/spark/pull/43310, the parquet supported compression codecs and spark supported are not completely one-on-one.

Reply:
Ok, looks fine